src/meta/MetaHttpIngestHandler.cpp

```diff
@@ -0,0 +1,168 @@
+            return;
+        default:
+            break;
+    }
+
+    if (ingestSSTFiles(space_, path_)) {
+        LOG(INFO) << "SSTFile ingest successfully " << path_;
+        ResponseBuilder(downstream_)
+            .status(WebServiceUtils::to(HttpStatusCode::OK),
+                    WebServiceUtils::toString(HttpStatusCode::OK))
+            .body("SSTFile ingest successfully")
+            .sendWithEOM();
+    } else {
+        LOG(ERROR) << "SSTFile ingest failed";
+        ResponseBuilder(downstream_)
+            .status(WebServiceUtils::to(HttpStatusCode::FORBIDDEN),
+                    WebServiceUtils::toString(HttpStatusCode::FORBIDDEN))
+            .body("SSTFile ingest failed")
+            .sendWithEOM();
+    }
+}
+
+void MetaHttpIngestHandler::onUpgrade(UpgradeProtocol) noexcept {
+    // Do nothing
+}
+
+
+void MetaHttpIngestHandler::requestComplete() noexcept {
+    delete this;
+}
+
+
+void MetaHttpIngestHandler::onError(ProxygenError error) noexcept {
+    LOG(ERROR) << "Web Service MetaHttpIngestHandler got error : "
+               << proxygen::getErrorString(error);
+}
+
+bool MetaHttpIngestHandler::ingestSSTFiles(GraphSpaceID space, const std::string& path) {
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto prefix = MetaServiceUtils::partPrefix(space);
+    auto ret = kvstore_->prefix(0, 0, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        LOG(ERROR) << "Fetch Parts Failed";
+        return false;
+    }
+
+    std::set<std::string> storageIPs;
+    while (iter->valid()) {
+        for (auto host : MetaServiceUtils::parsePartVal(iter->val())) {
+            auto storageIP = network::NetworkUtils::intToIPv4(host.get_ip());
+            if (std::find(storageIPs.begin(), storageIPs.end(), storageIP) == storageIPs.end()) {
+                storageIPs.insert(std::move(storageIP));
+            }
+        }
+        iter->next();
+    }
+
+    std::vector<folly::SemiFuture<bool>> futures;
+    static nebula::thread::GenericThreadPool pool;
+    std::call_once(ingestPoolStartFlag, []() {
+        LOG(INFO) << "Ingest Thread Pool start";
+        pool.start(FLAGS_meta_ingest_thread_num);
+    });
+
+    for (auto &storageIP : storageIPs) {
+        auto dispatcher = [storageIP, space, path]() {
+            auto tmp = "http://%s:%d/ingest?path=%s&space=%d";
+            auto url = folly::stringPrintf(tmp, storageIP.c_str(), FLAGS_storage_ingest_http_port,
+                                           path.c_str(), space);
+            auto command = folly::stringPrintf("/usr/bin/curl -G \"%s\"", url.c_str());
+            LOG(INFO) << "Command: " << command;
+            auto ingestResult = ProcessUtils::runCommand(command.c_str());
+            return ingestResult.ok() && ingestResult.value() == "SSTFile ingest successfully";
+        };
+        auto future = pool.addTask(dispatcher);
+        futures.push_back(std::move(future));
+    }
+
+    std::atomic<bool> successfully{true};
```

Comment:
here no need atomic, the wait() will synchronize the pool and current thread to access successfully